src/Sentry/Laravel/Tracing/Middleware.php

```diff
@@ -91,29 +91,29 @@ private function startTransaction(Request $request, HubInterface $sentry): void
             'url' => '/' . ltrim($request->path(), '/'),
             'method' => strtoupper($request->method()),
         ]);
-        $context->setStartTimestamp($request->server('REQUEST_TIME_FLOAT', $fallbackTime));
+        $context->setStartTimestamp($requestStartTime);
 
         $this->transaction = $sentry->startTransaction($context);
 
         // Setting the Transaction on the Hub
         SentrySdk::getCurrentHub()->setSpan($this->transaction);
 
         if (!$this->addBootTimeSpans() && app() instanceof Laravel) {
-            // @TODO: We might want to move this together with the `RouteMatches` listener to some central place and or do this from the `EventHandler`
-            app()->booted(function () use ($request, $fallbackTime): void {
+            $laravelStartTime = defined('LARAVEL_START') ? LARAVEL_START : $request->server('REQUEST_TIME_FLOAT');
+            if ($laravelStartTime !== null) {
                 $spanContextStart = new SpanContext();
                 $spanContextStart->setOp('app.bootstrap');
-                $spanContextStart->setStartTimestamp(defined('LARAVEL_START') ? LARAVEL_START : $request->server('REQUEST_TIME_FLOAT', $fallbackTime));
+                $spanContextStart->setStartTimestamp($laravelStartTime);
                 $spanContextStart->setEndTimestamp(microtime(true));
                 $this->transaction->startChild($spanContextStart);
+            }
```

Comment:
Note, this is a change compared to the initial implementation and I am open to be challenged on this assumption here, but, if neither `LARAVEL_START` constant is defined nor `REQUEST_TIME_FLOAT` SAPI is available, we should not just use timestamp of the start of this middleware, because the difference between these two values will be negligible and will misreport the "bootstrap" and will be misleading for those who observing timings, making them think that bootstrap was actually nearly instantaneous, while that is not even remotely true.

Reply:
Thinking about this some more and you are right, however I feel like the bootstrap section is not being made longer then it needs to be too since ending up in `startTransaction` is not the actual `booted` time of the app but a little after it.

Since the booted event happens before the transaction is even started we would have to buffer it. I might be going to say some dirty dirty words here but what if we would add a booted event callback in the event listener and set a static property on the middleware to have the actual timestamp of when the booted event occurred we can use here to set the actual end timestamp, the start timestamp is fine like how you did it I feel.

We can also then set the start for the `app.handle` frame to the end timestamp of the bootstrap section since they come right after each other.

Reply:
The static property would actually work... but we can also have a shared stateful dependency that's injected into the middleware and used by the callback to achieve this same result without it being too "dirty". Happy to do that if you also think that is by any means better.

As to:
> We can also then set the start for the `app.handle` frame to the end timestamp of the bootstrap section since they come right after each other.

This would actually break things for long running processes, because we bootstrap once and then handle request from then on, and I think it is better to have transaction start time originating from middleware. What do you say?